### PR TITLE
Cyril/add loading spinner

### PR DESCRIPTION
**Pull request by cyrielo**
# Description
This PR add a new loading spinner to the page, when a user controls the filter or the sorting

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?
- [x] Header page
- [x] Report Page

Specific components 

* Header component
* ReportFilter component
* PipelineSample Report

### Report Page
* Data loads succesfully 
* Collapse and Uncollapse button works
* Clicking on the taxonomy name gives you total reads for that taxon
* Category filter works
* Genus Search works
* Filters work as expected
* Disabling Filters works
* Filtering by column works 

# Checklist:

- [x] I have run through the testing script to make sure current functionality is unchanged
- [x] I have done relevant tests that prove my fix is effective or that my feature works
- [x] I have spent time testing out edge cases for my feature
- [x] I have updated the test script or pull request template if necessary
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/assets/javascripts/components/ReportFilter.jsx [hunk 1]
```diff
@@ -34,14 +34,23 @@ class ReportFilter extends React.Component {
...
+    $('.page-loading').css('display', 'flex');
+  }
+
   applyExcludedCategories(e) {
+    ReportFilter.showLoading('Applying category filter...');
     this.props.applyExcludedCategories(e.target.value, e.target.checked)
   }
 
   applyGenusFilter(selected_genus) {
+    ReportFilter.showLoading(`Now filtering by '${selected_genus}'`);
```
Comment (cdebourcy) [inline app/assets/javascripts/components/ReportFilter.jsx]: Should probably be `Now filtering for '${selected_genus}'`

File: app/assets/javascripts/components/PipelineSampleReport.jsx [hunk 2]
```diff
@@ -25,16 +25,20 @@ class PipelineSampleReport extends React.Component {
...
   enableFilters() {
     disable_filters = 0;
+    ReportFilter.showLoading('Enabling filters...');
     this.refreshPage({disable_filters});
   }
 
   // applySort needs to be bound at time of use, not in constructor above
   applySort(sort_by) {
+    const regex = new RegExp('_', 'g');
+    ReportFilter.showLoading(`Sorting the table by ${sort_by.replace(regex, ' ')}`);
```
Comment (cdebourcy) [inline app/assets/javascripts/components/PipelineSampleReport.jsx]: add `...` for consistency of all the messages?

File: app/assets/javascripts/components/ReportFilter.jsx (hunk 1 above)
Comment (cdebourcy) [reply app/assets/javascripts/components/ReportFilter.jsx]: (not filtering BY)

Comment (cyrielo) [reply app/assets/javascripts/components/ReportFilter.jsx]: Sweet

**Pull request merged**